src/components/Buildbaron/BuildBaronTab.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { useQuery } from "@apollo/client";
+import { Tab } from "@leafygreen-ui/tabs";
+import BuildBaron from "components/Buildbaron/BuildBaron";
+import { BuildBaronQuery, BuildBaronQueryVariables } from "gql/generated/types";
+import { GET_BUILD_BARON } from "gql/queries";
+import { usePrevious } from "hooks";
+
+interface Props {
+  taskId: string;
+  execution: number;
+}
+
+export const BuildBaronTab: React.FC<Props> = ({ taskId, execution }) => {
+  const {
+    data: buildBaronData,
+    error: buildBaronError,
+    loading: buildBaronLoading,
+    refetch,
+  } = useQuery<BuildBaronQuery, BuildBaronQueryVariables>(GET_BUILD_BARON, {
+    variables: { taskId, execution },
+  });
+
+  const prevExecution = usePrevious(execution);
+  if (execution !== prevExecution) {
+    refetch();
+  }
```

Comment:
I dont think this is necessary if you just place execution in a useeffect hook it should do the same thing.

```
useEffect(() => refetch(), [execution])
```